app/models/course_data/assignment.rb

```diff
@@ -105,6 +108,19 @@ def peer_review_pagename
     "#{sandbox_pagename}/#{user.username}_Peer_Review"
   end
 
+  def update_sandbox_url(new_url)
+    return unless new_url
+    return unless user
+    # creates and initializes instance variable (edit_sandbox_url)
+    # used to bypass the before_save (set_sandbox_url) method.
+    @edit_sandbox_url = true
+    language = wiki.language || 'www'
+    project = wiki.project || 'wikipedia'
+    base_url = "https://#{language}.#{project}.org/wiki"
+    new_sandbox_url = "#{base_url}/User:#{new_url}/#{article_title}"
```

Comment:
I don't think we want to assume that `new_url` is a username. I think the expected behavior would be to take a complete URL, perhaps validate it to make sure it's from the expected wiki, and then save it. It might or might not include the same article_title as the Assignment record.

Reply:
sure